plenum/test/replica/test_on_propagate_primary_done.py

```diff
@@ -0,0 +1,46 @@
+import sys
+
+
+def test_is_Master_update_watermarks(replica):
```

Comment:
Can we extend the tests to emulate the following situations:
1) catchup without view change when there is no primary (expected [0, 300])
- catchup_clear_for_backup
- _reset_watermarks_before_new_view
- _setup_for_non_master_after_view_change
2) catchup without view change when there is a primary (expected [0, max])
- catchup_clear_for_backup
3) view change (no propagate primary): expected [0, 300]
- catchup_clear_for_backup
- _reset_watermarks_before_new_view
- _setup_for_non_master_after_view_change
4) view change (propagate primary): expected [0, max] for backup and [0, 300] for master
- catchup_clear_for_backup
- _reset_watermarks_before_new_view
- _setup_for_non_master_after_view_change
- on_propagate_primary_done

We can emulate finishing catchup by calling